ADR/0023-git-references-to-furnish-integration-test-scenarios.md

```diff
@@ -0,0 +1,76 @@
+# 23. Git references to furnish Integration Test Scenarios
+
+* Date Documented: 2023-04-10
+* Date Accepted: TBD
+
+## Status
+
+In Review
+
+## Context
+
+Up to now, the Integration service has only supported setting the Tekton bundle (Bundle string `json:"bundle"`) in the 
+IntegrationTestScenario [CR] as a reference for the integration tests, in order to run the Tekton PipelineRuns.
+
+Below is the existing format to supply the Integration test references:
+
+![](../diagrams/ADR-0023/tekton-bundle-ITS.jpg)
+
+Building and maintaining Tekton bundles is cumbersome for users when providing custom tests.
+
+Resolving tasks from a git repository is more natural for developers and allows for faster iteration. Hence, users
+should be able to reference their Tekton pipeline definitions that are hosted in git repositories ([RHTAP-402]).
+
+To support the functionality of using git references, Tekton resolvers need to be enabled. As part of [PLNSRVCE-1030], 
+pipeline service adds support for the [Tekton-resolvers] including git resolvers which should in turn give us the ability
+to expand the support for resolvers in the IntegrationTestScenarios.
+
+## Decision
+
+Integration service now migrates to the new version of the IntegrationTestScenario CRD having the expanded functionality
+to allow using different types of Tekton resolvers ex: bundles-resolver, cluster-resolver, hub-resolver along with 
+the git-resolver. This gives the extensibility to permit the use of different resolvers in the future.
+
+The new format of adding the Integration Test Scenarios would look like:
+
+
+![](../diagrams/ADR-0023/git-references-ITS.jpg)
+
+Users git repository path to their test code can now provide the information or the location of their IntegrationTestScenarios
+through Git references like **Github URL**, **Git Options** (**Revision** & **Path in repository**). 
+
+The information provided by the users in the above form will then be consumed by the integration service with the help of IntegrationTestScenario CR and will be used to run the integration tests against the Application Snapshot in question.
+
+Example of the tekton pipeline definition with git resolver:
+
+![](../diagrams/ADR-0023/tekton-pipeline-definition-git-resolver.jpg)
+
+The existing IntegrationTestScenarios will be migrated to the new API version with tekton resolvers, by using the 
+standard kubernetes conversion webhooks ([STONEINTG-386]).
+
+## Consequences
```

Comment:
Can we address the impact of imposter git commits (`https://github.com/tektoncd/pipeline/issues/6315`)? Do we have any plan to help protect (or warn) users against them?

Reply:
I don't think we have it covered under the ongoing feature [RHTAP-402](https://issues.redhat.com/browse/RHTAP-402). Would you mind filing a jira as a follow-up for the improvements ?

Reply:
Using either open questions as in [ADR 3](https://github.com/redhat-appstudio/book/blob/main/ADR/0003-interacting-with-internal-services.md#open-questions) or out of scope as in [ADR 21](https://github.com/redhat-appstudio/book/blob/main/ADR/0021-partner-tasks.md#out-of-scope) would work for covering in this AR.

Reply:
This looks to me like a Tekton issue that will need to be handled in Tekton and then adopted by the Integration service if there are changes that need to be addressed. It feels like it could go to open questions, but specifying that the Integration service would adopt future security improvements from Tekton as soon as they become available. 
Does that sound good @sonam1412 @arewm?

Reply:
I agree that it is a Tekton issue, but this is also the first time that we are exposing the ability for users to be faced with this type of situation. We should be up front with the fact if we are not doing anything additional to mitigate this risk. Any documentation should also make sure to mention this potential attack vector. 

@sonam1412 shared a proposed wording with me, I assume it just needs to be committed to address this.

Reply:
Added it [here](https://github.com/redhat-appstudio/book/pull/92/files#diff-4adb99d2e0cdfd21098edf4375e38cd0ca48caeb8f285113121a39634a3989abR51-R56). 
Please take a look.

Reply:
Ack, looks good to me.